src/libUtils/Logger.h

```diff
@@ -160,25 +164,26 @@ class ScopeMarker {
   ~ScopeMarker();
 };
 
-#define INIT_FILE_LOGGER(fname_prefix) Logger::GetLogger(fname_prefix, true)
-#define INIT_STDOUT_LOGGER() Logger::GetLogger(NULL, false)
-#define INIT_STATE_LOGGER(fname_prefix) \
-  Logger::GetStateLogger(fname_prefix, true)
-#define INIT_EPOCHINFO_LOGGER(fname_prefix) \
-  Logger::GetEpochInfoLogger(fname_prefix, true)
+#define INIT_FILE_LOGGER(fname_prefix, logpath) \
+  Logger::GetLogger(fname_prefix, true, logpath)
+#define INIT_STDOUT_LOGGER() Logger::GetLogger(NULL, false, "./")
+#define INIT_STATE_LOGGER(fname_prefix, logpath) \
+  Logger::GetStateLogger(fname_prefix, true, logpath)
+#define INIT_EPOCHINFO_LOGGER(fname_prefix, logpath) \
+  Logger::GetEpochInfoLogger(fname_prefix, true, logpath)
 #define LOG_MARKER() ScopeMarker marker(__LINE__, __FILE__, __FUNCTION__)
-#define LOG_STATE(msg)                                                \
-  {                                                                   \
-    std::ostringstream oss;                                           \
-    auto cur = std::chrono::system_clock::now();                      \
-    auto cur_time_t = std::chrono::system_clock::to_time_t(cur);      \
-    oss << "[ " << std::put_time(gmtime(&cur_time_t), "%y-%m-%dT%T.") \
-        << PAD(get_ms(cur), 3, '0') << " ]" << msg;                   \
-    Logger::GetStateLogger(NULL, true).LogState(oss.str().c_str());   \
+#define LOG_STATE(msg)                                                    \
+  {                                                                       \
+    std::ostringstream oss;                                               \
+    auto cur = std::chrono::system_clock::now();                          \
+    auto cur_time_t = std::chrono::system_clock::to_time_t(cur);          \
+    oss << "[ " << std::put_time(gmtime(&cur_time_t), "%y-%m-%dT%T.")     \
+        << PAD(get_ms(cur), 3, '0') << " ]" << msg;                       \
+    Logger::GetStateLogger(NULL, true, "./").LogState(oss.str().c_str()); \
```

Comment:
All of these loggers that pass in `"./"` will be logging to a different path than the one passed in as an argument it seems. Is that the desired functionality or do you want all loggers to be logging to the custom path?

Reply:
Actually inside `GetStateLogger()` there is a static construction of `Logger` (singleton), while the fist construction call of `Logger` would be called by `INIT_STATE_LOGGER` within customized log path beforehand. The reason we still put `./` here is just in case the false usage forgetting to call `INIT_STATE_LOGGER` beforehand, then the default log path `./` should be used instead.